libautoscoper/src/Tracker.cpp

```diff
@@ -979,13 +983,12 @@ bool Tracker::calculate_viewport(const CoordFrame& modelview, const Camera& came
 // Save Full DRR Image
 void Tracker::getFullDRR(unsigned int volumeID) const
 {
-    double xyzypr[6] = { (*(const_cast<Trial&>(trial_)).getXCurve(-1))(trial_.frame),
-        (*(const_cast<Trial&>(trial_)).getYCurve(-1))(trial_.frame),
-        (*(const_cast<Trial&>(trial_)).getZCurve(-1))(trial_.frame),
-        (*(const_cast<Trial&>(trial_)).getYawCurve(-1))(trial_.frame),
-        (*(const_cast<Trial&>(trial_)).getPitchCurve(-1))(trial_.frame),
-        (*(const_cast<Trial&>(trial_)).getRollCurve(-1))(trial_.frame) };
-
+  float x_val = (*(const_cast<Trial&>(trial_)).getXCurve(-1))(trial_.frame);
+  float y_val = (*(const_cast<Trial&>(trial_)).getYCurve(-1))(trial_.frame);
+  float z_val = (*(const_cast<Trial&>(trial_)).getZCurve(-1))(trial_.frame);
+  Quatf quat_val = (*(const_cast<Trial&>(trial_)).getQuatCurve(-1))(trial_.frame);
+  Vec3f eulers = quat_val.toEuler();
+  double xyzypr[6] = { x_val, y_val, z_val, eulers.z, eulers.y, eulers.x };
   CoordFrame xcframe = CoordFrame::from_xyzypr(xyzypr);
```

Comment:
Consider introducing `CoordFrame::from_xyzq` to simplify, you would then simply do:

```
CoordFrame xcframe = CoordFrame::from_xyzq(x_val, y_Val, z_val, quat_val);
```